src/routers/semester.ts

```diff
@@ -1,24 +1,53 @@
 import express from 'express';
+import { create, read, update, del } from '../operators/semesterOperations';
 
 const semesterRouter = express.Router();
 
-semesterRouter.get('/', (req, res) => {
-	console.log('Get Semester');
-	res.send('Get semesterRouter');
+semesterRouter.get('/', async (req, res) => {
+	try {
+		console.log('Get Semester');
+		const semester = await read(req.body.id);
```

Comment:
So the HTTP standard says that we should not use the request body in a get request [see here](https://stackoverflow.com/questions/978061/http-get-with-request-body). So instead we should use a parameter:

```ts
semesterRouter.get('/:id', async (req, res) => {
        req.params.id; // This is how you access.
}
```

[Relevant express docs](https://expressjs.com/en/guide/routing.html#route-parameters)


This only applies to the get request, I think everything else can use the body, although if we wanted we could use this for all the paths to single out the id.

Reply:
Does this mean our get requests will not contain a body and instead the user id will be contained in the request url?

For example:

`Route path: /:id
Request URL: http://localhost:3000/uniqueIdHere
req.params: { "id": "uniqueIdHere" }`

Reply:
Yes like that except remember the routes you create are prefixed with the data type (already done in another file): https://localhost:3000/semester/uuid-here and available still like you mention

Reply:
ok, perfect - I'll update this function!